paasta_tools/docker_wrapper.py

```diff
@@ -134,6 +138,51 @@ def can_add_mac_address(args):
     return True
 
 
+def can_add_ip_address(args):
+    # Can only specify an IP address for user defined networks
+    if is_network_host(args) or not is_run(args):
+        return False
+
+    for index, arg in enumerate(args):
+        # Check for --ip
+        if arg.startswith('--ip'):
+            return False
+
+    return True
+
+
+def get_ip_for_container():
+    # This would look at the currently outstanding IP addresses
+    # Something like but not necessarily =>
+    current_ip_addresses = subprocess.check_output((
```

Comment:
I feel like this is certainly better to be done in python, but I had this handy and I don't know if the 3 year old, 2 major versions behind of `docker-py` the repo pins to has this functionality.